gips/data/merra/merra.py

```diff
@@ -220,40 +224,48 @@ def __init__(self, filename):
         self.date = datetime.datetime.strptime(parts[3], '%Y%j').date()
         self.products[self.asset] = filename
 
-
     @classmethod
     def opendap_fetch(cls, asset, date):
         """ Get array proxy from OpenDap for this asset and date """
+
         url = cls._assets[asset].get('url', '')
         if url == '':
             raise Exception("%s: URL not defined for asset %s" % (cls.__name__, asset))
-        success = False
-
-        for ver in ['100', '200', '300', '301', '400']:
 
+        # TODO: instead of trying all possible vers, get the number from XML metadata
+        success = False
+        for ver in ['100', '200', '300', '301', '400']:            
             if asset != "FRLAND":
                 f = cls._assets[asset]['source'] % (ver, date.year, date.month, date.day)
                 loc = "%s/%04d/%02d/%s" % (url, date.year, date.month, f)
             else:
                 f = cls._assets[asset]['source'] % (ver, 0, 0, 0)
                 loc = "%s/1980/%s" % (url, f)
+            
             try:
-                with Timeout(30):
+                with Timeout(30):          
                     dataset = open_url(loc)
             except Timeout.Timeout:
                 # change to verbose_out()
                 print "Timeout"
+            except urllib2.HTTPError:
+                # TODO: see above
+                pass
```

Comment:
This is technically an improvement on what was here before, so fine, but as long as you're editing things, it should be explicitly explained above that it's passing errors in silence because it believes it has other URLs to try.